overlord/configstate/configcore/tmpfs.go

```diff
@@ -0,0 +1,127 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/sysconfig"
+)
+
+const (
+	mntStaticOptions         = "mode=1777,strictatime,nosuid,nodev"
+	tmpfsService             = "tmp.mount"
+	tmpfsMountPoint          = "/tmp"
+	tmpMntServOverrideSubDir = "tmp.mount.d"
+	tmpMntServOverrideFile   = "override.conf"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.tmpfs.size"] = true
+}
+
+// Regex matches what is specified by tmpfs(5) for the size option
+var validTmpfsSize = regexp.MustCompile(`^[0-9]+[kmgKMG%]?$`).MatchString
+
+func validateTmpfsSettings(tr config.ConfGetter) error {
+	tmpfsSz, err := coreCfg(tr, "tmpfs.size")
+	if err != nil {
+		return err
+	}
+	if tmpfsSz == "" {
+		return nil
+	}
+	if !validTmpfsSize(tmpfsSz) {
+		return fmt.Errorf("cannot set tmpfs size %q: invalid size", tmpfsSz)
+	}
```

Comment:
I wonder if we should enforce a minimal size here too as part of the validation? I suspect setting tmpfize to 10byte or something may even break the boot? so maybe have a minimal size here makes sense? maybe 5% of the osutil.TotalUsableMemory() or something? Or 16mb? no strong opinion.

Reply:
This is done now, using 16Mb, which I think is a reasonable minimum. Normal boot works fine with that, although big snaps will not install. I guess we should use `/var/tmp` on snap installations though, although that is a different topic.

Reply:
regarding using /var/tmp, we already do use /var/lib/snapd/... for both store and locally installed snaps, we did recently have a bug that was fixed where Go's standard library would also try to cache files in /tmp too, and we had to work around that as well

Reply:
Nice! I knew there were problems in the past, great to see those are already fixed.